### PR TITLE
cleanup dependency and remove asm handled by gizmo

### LINKED ISSUE #224: ASM dependency?
(Maybe GH discussions should be enabled for this repo, which is now possible as an opt-in for quarkiverse projects.)

I noticed that this extension brings in `org.ow2.asm:asm:jar:9.1:compile`. But I'd assume that all bytecode magic is done during build time instead of runtime?

### PR DESCRIPTION
**Pull request by dufoli**
fix #224

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -109,7 +115,18 @@
                 <groupId>jakarta.xml.ws</groupId>
                 <artifactId>jakarta.xml.ws-api</artifactId>
                 <version>${jakarta.xml.ws-api.version}</version>
-            </dependency>     
+            </dependency>
+            <dependency>
+                <groupId>org.glassfish.jaxb</groupId>
+                <artifactId>jaxb-runtime</artifactId>
```
Comment (famod) [inline pom.xml:121]: There is also `quarkus-jaxb`... 🤔

Comment (shumonsharif) [reply pom.xml]: Good point @famod we already have a dependency on quarkus-jaxb, which has everything we need; I have done some additional cleanup.

**Comment by shumonsharif** [comment]
@dufoli @famod Please review latest updates.

**Comment by dufoli** [comment]
@shumonsharif  Maybe add quarkus-jaxb to main pom in dependencyManagement to have all external dep in main pom wdyt ?. else it is ok for me.

**Comment by shumonsharif** [comment]
Hi thanks @dufoli the `quarkus-bom` dependency in the main pom in dependencyManagement is what makes `quarkus-jaxb` available.  Even if we define in dependencyManagement, we'll still need it also in the `runtime/pom.xml` as well.

**Review by famod — APPROVED** [review]
LGTM!

**Comment by famod** [comment]
Btw, if you want to make extra sure this doesn't slip in again, then you could add something like https://maven.apache.org/enforcer/enforcer-rules/bannedDependencies.html

I would also suggest https://maven.apache.org/enforcer/enforcer-rules/dependencyConvergence.html but that addresses another category of problems.

**Pull request merged**